src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -423,17 +428,35 @@ private void handleResponse(BulkItemResponse response, long executionId) {
           return;
         }
       }
-
       if (response.getFailureMessage().contains(VERSION_CONFLICT_EXCEPTION)) {
-        log.warn(
-            "Ignoring version conflict for operation {} on document '{}' version {} in index '{}'.",
-            response.getOpType(),
-            response.getId(),
-            response.getVersion(),
-            response.getIndex()
+        // For external version conflicts, response.getVersion() will be returned as -1,
+        // but we have the actual version number for this record because we set it in
+        // the request.
+        long thisRecordVersion = request.versionType() == VersionType.EXTERNAL
+                ? request.version() : response.getVersion();
+        String message = String.format("Ignoring %s version conflict for operation %s on "
```

Comment:
Performance Nit: we have seen millions of records going to DLQ and VersionType is internal then we would do this string concatenation only to do log.debug

Should we instead extract just the template ` "Ignoring version conflict for operation {} on document '{}' version {} in index '{}'."`  to a variable and call it using log.warn and log.debug with replacement params

Reply:
Sorry, not sure what you are suggesting. Can you please clarify?

Reply:
Something like

```
String message ="Ignoring {} version conflict for operation {} on document '{}' version {} in index '{}'.";
if (request.versionType() != VersionType.EXTERNAL) {
          log.warn(message,  request.versionType(),
                response.getOpType(),
                response.getId(),
                thisRecordVersion,
                response.getIndex());
                ........
} else {
          log.debug(message,  request.versionType(),
                response.getOpType(),
                response.getId(),
                thisRecordVersion,
                response.getIndex());	
}

```